integ-test/src/test/java/org/opensearch/sql/sql/PaginationFilterIT.java

```diff
@@ -0,0 +1,152 @@
+        testData.add(new Object[] { statement, totalHits, pageSize });
+      }
+    });
+    return testData;
+  }
+
+  /**
+   * Test compares non-paginated results with paginated results
+   * To ensure that the pushdowns return the same number of hits even
+   * with filter WHERE pushed down
+   */
+  @Test
+  @SneakyThrows
+  public void test_pagination_blackbox() {
+    // get non-paginated result for comparison
+    JSONObject nonPaginatedResponse = executeJdbcRequest(sqlStatement);
+    int totalResultsCount = nonPaginatedResponse.getInt("total");
+    JSONArray rows = nonPaginatedResponse.getJSONArray("datarows");
+    JSONArray schema = nonPaginatedResponse.getJSONArray("schema");
+    var testReportPrefix = String.format("query: %s; total hits: %d; page size: %d || ", sqlStatement, totalResultsCount, pageSize);
+    assertEquals(totalHits.intValue(), totalResultsCount);
+
+    var rowsPaged = new JSONArray();
+    var pagedSize = 0;
+    var responseCounter = 1;
+
+    // make first request - with a cursor
+    JSONObject paginatedResponse = new JSONObject(executeFetchQuery(sqlStatement, pageSize, "jdbc"));
+    this.logger.info(testReportPrefix + "<first response>");
+    while (paginatedResponse.has("cursor")) {
+      var cursor = paginatedResponse.getString("cursor");
+      pagedSize += paginatedResponse.getInt("size");
+      var datarows = paginatedResponse.getJSONArray("datarows");
+      for (int i = 0; i < datarows.length(); i++) {
+        rowsPaged.put(datarows.get(i));
+      }
+
+      assertEquals(totalResultsCount, paginatedResponse.getInt("total"));
+      assertTrue(
+          "Paged response schema doesn't match to non-paged",
+          schema.similar(paginatedResponse.getJSONArray("schema")));
+      assertTrue(
+          testReportPrefix + "Cursor returned from legacy engine",
+          cursor.startsWith("n:"));
+
+      paginatedResponse = executeCursorQuery(cursor);
+      this.logger.info(testReportPrefix
+          + String.format("response %d/%d", responseCounter++, (totalResultsCount / pageSize) + 1));
+    }
+    // last page expected results:
+    assertEquals(0, paginatedResponse.getInt("total"));
+    assertEquals(testReportPrefix + "Last page is not empty",
+        0, paginatedResponse.getInt("size"));
+    assertEquals(testReportPrefix + "Last page is not empty",
+        0, paginatedResponse.getJSONArray("datarows").length());
+
+    // compare paginated and non-paginated counts
+    assertEquals(testReportPrefix + "Paged responses returned an unexpected total",
+        totalResultsCount, pagedSize);
+    assertEquals(testReportPrefix + "Paged responses returned an unexpected rows count",
+        rows.length(), rowsPaged.length());
+  }
+
+  // A dummy class created, because accessing to `client()` isn't available from a static context,
+  // but it is needed before an instance of `PaginationBlackboxIT` is created.
+  private static class PaginationBlackboxHelper extends SQLIntegTestCase {
```

Comment:
Is this class necessary?

I don't see where `client()` is used or understand how having this class helps with accessing it.

Reply:
removing